### PR TITLE
Correctly calculate resample for mapcube plotting.

### PR DESCRIPTION
**Pull request by Cadair**

### CODE REVIEW

File: sunpy/map/mapcube.py [hunk 1]
```diff
@@ -192,10 +193,13 @@ def annotate_frame(i):
...
         if resample:
-            # This assumes that the maps are homogeneous!
-            # TODO: Update this!
-            resample = np.array(len(self.maps)-1) * np.array(resample)
-            ani_data = [x.resample(resample) for x in self.maps]
+            if self.all_maps_same_shape():
+                resample = u.Quantity(self.maps[0].dimensions) * np.array(resample)
+                ani_data = []
+                for amap in self.maps:
+                    ani_data.append(amap.resample(resample))
```
Comment (dpshelio) [inline sunpy/map/mapcube.py]: why this cannot be done like before?
```python
ani_data  = [ amap.resample(resample) for amap in self.maps ]
```

Comment (Cadair) [inline sunpy/map/mapcube.py]: no reason I guess.

Comment (Cadair) [inline sunpy/map/mapcube.py]: Think I adapted it from below with the `.append`

**Comment by dpshelio** [comment]
:+1:

**Pull request merged**